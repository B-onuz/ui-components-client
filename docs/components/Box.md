# Box (Component)

Componente de caixa para textos, tabelas e cards.

## Como utilizar

```js
import Box from 'ui-components/lib/components/Box'
```

```jsx
<Box>
  <h1>Dolor deserunt et dolore cupidatat.</h1>
  <p>Quis tempor veniam cillum cillum enim deserunt ullamco exercitation.</p>
</Box>
```

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
