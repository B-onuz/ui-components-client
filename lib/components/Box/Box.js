"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Box/Box.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  box-shadow: ", ";\n  background: ", ";\n  padding: ", ";\n  border-radius: 10px;\n  display: ", ";\n  flex: ", ";\n  justify-content: ", ";\n  flex-direction: ", ";\n  align-items: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledBox = _styledComponents["default"].div(_templateObject(), (0, _styledSystem.compose)(_styledSystem.space, _styledSystem.layout), function (_ref) {
  var noBorder = _ref.noBorder;
  return noBorder ? 'none' : '0px 3px 6px #2c282846';
}, function (_ref2) {
  var noBorder = _ref2.noBorder;
  return noBorder ? 'none' : '#ffffff 0% 0% no-repeat padding-box';
}, function (_ref3) {
  var noBorder = _ref3.noBorder;
  return noBorder ? '0' : '10px';
}, function (_ref4) {
  var display = _ref4.display;
  return display;
}, function (_ref5) {
  var flex = _ref5.flex;
  return flex;
}, function (_ref6) {
  var justifyContent = _ref6.justifyContent;
  return justifyContent;
}, function (_ref7) {
  var flexDirection = _ref7.flexDirection;
  return flexDirection;
}, function (_ref8) {
  var alignItems = _ref8.alignItems;
  return alignItems;
});

var Box = function Box(_ref9) {
  var display = _ref9.display,
      justifyContent = _ref9.justifyContent,
      flexDirection = _ref9.flexDirection,
      alignItems = _ref9.alignItems,
      flex = _ref9.flex,
      noBorder = _ref9.noBorder,
      rest = _objectWithoutPropertiesLoose(_ref9, ["display", "justifyContent", "flexDirection", "alignItems", "flex", "noBorder"]);

  return /*#__PURE__*/_react["default"].createElement(StyledBox, _extends({
    noBorder: !!noBorder,
    display: display,
    justifyContent: justifyContent,
    flexDirection: flexDirection,
    alignItems: alignItems,
    flex: flex
  }, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 20,
      columnNumber: 5
    }
  }));
};

var _default = Box;
exports["default"] = _default;
module.exports = exports.default;