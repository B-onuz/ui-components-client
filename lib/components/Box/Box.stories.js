'use strict'

function _typeof(obj) {
  '@babel/helpers - typeof'
  if (typeof Symbol === 'function' && typeof Symbol.iterator === 'symbol') {
    _typeof = function _typeof(obj) {
      return typeof obj
    }
  } else {
    _typeof = function _typeof(obj) {
      return obj && typeof Symbol === 'function' && obj.constructor === Symbol && obj !== Symbol.prototype ? 'symbol' : typeof obj
    }
  }
  return _typeof(obj)
}

exports.__esModule = true
exports.EmptyTableNoBorderBoxExample = exports.BoxExample = exports['default'] = void 0

var _react = _interopRequireWildcard(require('react'))

var _Box = _interopRequireDefault(require('./Box'))

var _Table = _interopRequireDefault(require('../Table'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Box/Box.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

function _getRequireWildcardCache() {
  if (typeof WeakMap !== 'function') return null
  var cache = new WeakMap()
  _getRequireWildcardCache = function _getRequireWildcardCache() {
    return cache
  }
  return cache
}

function _interopRequireWildcard(obj) {
  if (obj && obj.__esModule) {
    return obj
  }
  if (obj === null || (_typeof(obj) !== 'object' && typeof obj !== 'function')) {
    return { default: obj }
  }
  var cache = _getRequireWildcardCache()
  if (cache && cache.has(obj)) {
    return cache.get(obj)
  }
  var newObj = {}
  var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor
  for (var key in obj) {
    if (Object.prototype.hasOwnProperty.call(obj, key)) {
      var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null
      if (desc && (desc.get || desc.set)) {
        Object.defineProperty(newObj, key, desc)
      } else {
        newObj[key] = obj[key]
      }
    }
  }
  newObj['default'] = obj
  if (cache) {
    cache.set(obj, newObj)
  }
  return newObj
}

var _default = {
  title: 'Box',
  component: _Box['default'],
}
exports['default'] = _default

var BoxExample = function BoxExample() {
  var _useState = (0, _react.useState)('nome'),
    sort = _useState[0],
    setSort = _useState[1]

  var _useState2 = (0, _react.useState)('asc'),
    order = _useState2[0],
    setOrder = _useState2[1]

  var handleChangeOrder = function handleChangeOrder(item) {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return /*#__PURE__*/ _react['default'].createElement(
    _Box['default'],
    {
      width: 1000,
      px: 0,
      py: 37,
      m: 30,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 24,
        columnNumber: 5,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_Table['default'], {
      width: 1,
      maxHeight: 200,
      onChangeOrder: handleChangeOrder,
      pagination: {
        sort: sort,
        order: order,
      },
      headers: [
        {
          title: 'Advogados',
          key: 'nome',
          sort: true,
          align: 'left',
        },
        {
          title: 'Nome',
          key: 'lawyer.first',
          sort: true,
          align: 'left',
        },
        {
          title: 'Sobrenome',
          key: 'lawyer.last',
          sort: true,
          align: 'center',
        },
        {
          title: 'Ações',
          key: 'actions',
          align: 'center',
        },
      ],
      data: [
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
        {
          nome: 'Last Nome do advogado',
          lawyer: {
            first: 'Jhon',
            last: 'Silva',
          },
          tarefas: 10,
        },
      ],
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 25,
        columnNumber: 7,
      },
    })
  )
}

exports.BoxExample = BoxExample

var EmptyTableNoBorderBoxExample = function EmptyTableNoBorderBoxExample() {
  var _useState3 = (0, _react.useState)('nome'),
    sort = _useState3[0],
    setSort = _useState3[1]

  var _useState4 = (0, _react.useState)('asc'),
    order = _useState4[0],
    setOrder = _useState4[1]

  var handleChangeOrder = function handleChangeOrder(item) {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return /*#__PURE__*/ _react['default'].createElement(
    _Box['default'],
    {
      width: 1000,
      px: 0,
      py: 37,
      m: 30,
      noBorder: true,
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 94,
        columnNumber: 5,
      },
    },
    /*#__PURE__*/ _react['default'].createElement(_Table['default'], {
      width: 1,
      maxHeight: 200,
      onChangeOrder: handleChangeOrder,
      pagination: {
        sort: sort,
        order: order,
      },
      headers: [
        {
          title: 'Advogados',
          key: 'nome',
          sort: true,
          align: 'left',
        },
        {
          title: 'Nome',
          key: 'lawyer.first',
          sort: true,
          align: 'left',
        },
        {
          title: 'Sobrenome',
          key: 'lawyer.last',
          sort: true,
          align: 'center',
        },
        {
          title: 'Ações',
          key: 'actions',
          align: 'center',
        },
      ],
      data: [],
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 95,
        columnNumber: 7,
      },
    })
  )
}

exports.EmptyTableNoBorderBoxExample = EmptyTableNoBorderBoxExample
