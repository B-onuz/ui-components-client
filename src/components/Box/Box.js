import React from 'react'
import styled from 'styled-components'
import { space, layout, compose } from 'styled-system'

const StyledBox = styled.div`
  ${compose(space, layout)}
  box-shadow: ${({ noBorder }) => (noBorder ? 'none' : '0px 3px 6px #2c282846')};
  background: ${({ noBorder }) => (noBorder ? 'none' : '#ffffff 0% 0% no-repeat padding-box')};
  padding: ${({ noBorder }) => (noBorder ? '0' : '10px')};
  border-radius: 10px;
  display: ${({ display }) => display};
  flex: ${({ flex }) => flex};
  justify-content: ${({ justifyContent }) => justifyContent};
  flex-direction: ${({ flexDirection }) => flexDirection};
  align-items: ${({ alignItems }) => alignItems};
`

const Box = ({ display, justifyContent, flexDirection, alignItems, flex, noBorder, ...rest }) => {
  return (
    <StyledBox
      noBorder={!!noBorder}
      display={display}
      justifyContent={justifyContent}
      flexDirection={flexDirection}
      alignItems={alignItems}
      flex={flex}
      {...rest}
    />
  )
}

export default Box
