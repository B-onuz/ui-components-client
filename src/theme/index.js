const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
