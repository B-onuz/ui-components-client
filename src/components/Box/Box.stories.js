import React, { useState } from 'react'
import Box from './Box'
import Table from '../Table'

export default {
  title: 'Box',
  component: Box,
}

export const BoxExample = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <Box width={1000} px={0} py={37} m={30}>
      <Table
        width={1}
        maxHeight={200}
        onChangeOrder={handleChangeOrder}
        pagination={{
          sort,
          order,
        }}
        headers={[
          {
            title: 'Advogados',
            key: 'nome',
            sort: true,
            align: 'left',
          },
          {
            title: 'Nome',
            key: 'lawyer.first',
            sort: true,
            align: 'left',
          },
          {
            title: 'Sobrenome',
            key: 'lawyer.last',
            sort: true,
            align: 'center',
          },
          {
            title: 'Ações',
            key: 'actions',
            align: 'center',
          },
        ]}
        data={[
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
          { nome: 'Last Nome do advogado', lawyer: { first: 'Jhon', last: 'Silva' }, tarefas: 10 },
        ]}
      />
    </Box>
  )
}
export const EmptyTableNoBorderBoxExample = () => {
  const [sort, setSort] = useState('nome')
  const [order, setOrder] = useState('asc')

  const handleChangeOrder = (item) => {
    if (item.key === sort) {
      setOrder(order === 'asc' ? 'desc' : 'asc')
    } else {
      setSort(item.key)
      setOrder('asc')
    }
  }

  return (
    <Box width={1000} px={0} py={37} m={30} noBorder>
      <Table
        width={1}
        maxHeight={200}
        onChangeOrder={handleChangeOrder}
        pagination={{
          sort,
          order,
        }}
        headers={[
          {
            title: 'Advogados',
            key: 'nome',
            sort: true,
            align: 'left',
          },
          {
            title: 'Nome',
            key: 'lawyer.first',
            sort: true,
            align: 'left',
          },
          {
            title: 'Sobrenome',
            key: 'lawyer.last',
            sort: true,
            align: 'center',
          },
          {
            title: 'Ações',
            key: 'actions',
            align: 'center',
          },
        ]}
        data={[]}
      />
    </Box>
  )
}
