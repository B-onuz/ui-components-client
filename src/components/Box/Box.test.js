import React from 'react'
import Box from './Box.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('when render Box', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Box>
        <h1>Qui esse pariatur esse duis tempor enim dolor non ad culpa deserunt dolore culpa dolore.</h1>
        <p>Reprehenderit anim est officia veniam qui veniam aute sint dolore.</p>
      </Box>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
