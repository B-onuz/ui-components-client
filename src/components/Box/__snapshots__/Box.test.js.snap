// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when render Box 1`] = `
.c0 {
  box-shadow: 0px 3px 6px #2c282846;
  background: #ffffff 0% 0% no-repeat padding-box;
  padding: 10px;
  border-radius: 10px;
}

<div
  className="c0"
>
  <h1>
    Qui esse pariatur esse duis tempor enim dolor non ad culpa deserunt dolore culpa dolore.
  </h1>
  <p>
    Reprehenderit anim est officia veniam qui veniam aute sint dolore.
  </p>
</div>
`;
